### ISSUE TITLE
[xychart] make react-spring optional dependency

### ISSUE THREAD

**Issue by sachinraja** [airbnb/visx#1534]
I don't use any of the `Animated` components in my Chart, so I do not want to install `@react-spring/web`, which gives me a ton of errors with React 18. Could we change the xychart package so the animated components are exported from a separate entry point and `@react-spring/web` and `@visx/react-spring` are declared as optional peer dependencies?

**Comment by williaster** [comment]
Yup, this is a great idea. When we first published the package I think I spent some time trying to do just this but hit some issues. I can try and take a look this week but if anyone else wants to take a shot I'm happy to 👀  a PR as well 🙏

**Comment by williaster** [comment]
Noting this would likely be a breaking change for this package

**Comment by minut-shan** [comment]
@williaster Do you have any updates on this?
